### PR TITLE
Ol 5

### PR DESCRIPTION
**Pull request by will-moore**
This is a work in progress to update to latest OpenLayers 5.3.0, which doesn't use closure compiler so it doesn't need a separate build step.

At this point (3ab6501), I'm not aware of any new bugs. All functionality should be unchanged.

I have removed the closure compiler step from the npm build commands in ```package.json``` and they seem to work OK.

Pre-existing bugs (might be able to fix):
 - Double-click to complete Polyline/Polygon also zooms (pre-existing bug)?
 - Drag a Rectangle, then try to resize it - it remembers previous coordinates.

Other TODOs / questions:
 - Fix tests: ```ant unit-tests-only``` and reactivate in ```setup.py```
 - Remove ol3 references? E.g. rename ol3-viewer.js and replace "ol3" everywhere (lots).
 - Naming strategy etc. Viewer code is currently in ```/src/viewers/viewer/```.
 - Use OlMap instead of PluggableMap everywhere
 - Investigate why some imports are undefined. E.g. ```globals.PROJECTION``` is undefined when imported into ```utils/Misc```.
 - More build cleanup - remove anything required for google closure etc.
 - Investigate the docs build that Harald had. Do we want to maintain it etc? Lots of links to ```ome.ol3``` in the doc strings have now been removed etc.

### CODE REVIEW

**Comment by jburel** [comment]
@will-moore cf. build question, there is also work (onhold) done by @rgozim see https://github.com/ome/omero-iviewer/pull/215

**Comment by will-moore** [comment]
I think it will take me some time to understand the build and testing setup. I'm going to focus on the ROI fixes mentioned above first.

**Comment by will-moore** [comment]
Those last 2 commits fix the drawing of new shapes. Now off to buy new power cable....

File: plugin/setup.py [hunk 1]
```diff
@@ -78,7 +78,8 @@ def run(self):
 class Test(setuptools.command.install.install):
 
     def run(self):
-        self.spawn(['ant', 'unit-tests-only'])
+        # self.spawn(['ant', 'unit-tests-only'])
```
Comment (jburel) [inline plugin/setup.py]: add a TODO: reactivate tests

File: src/viewers/viewer/interaction/Select.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        /**
+         * @private
+         * @type {ol.interaction.SelectFilterFunction}
+         */
+        this.filter_ = TRUE;
+    };
+
+    /**
+     * Clears/unselects all selected features
+     *
```
Comment (jburel) [inline src/viewers/viewer/interaction/Select.js]: extra line

File: src/viewers/viewer/interaction/Select.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        return this.features_;
+    }
+
+    /**
+     * De/Selects a feature
+     *
+     * @param {ol.Feature} feature the feature to be (de)selected
+     * @param {boolean} select if true we want to select, otherwise deselect,
+     *                   the former being default
+     * @param {boolean=} remove_first on select we remove first to make sure that we
```
Comment (jburel) [inline src/viewers/viewer/interaction/Select.js]: extra ``=``

File: src/viewers/viewer/interaction/Translate.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                    var clonedF = f.clone();
+                    clonedF.setId(f.getId());
+                    this.translatedFeatures_.push(clonedF);
+                }
+            }, this);
+
+        // a listener to note whether there was an actual translate ...
+        listen(
+            this,
+            'translating',  // ol.interaction.TranslateEventType.TRANSLATING,
```
Comment (jburel) [inline src/viewers/viewer/interaction/Translate.js]: remove commented out code

File: src/viewers/viewer/interaction/Translate.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+                typeof hit['permissions']['canEdit'] === 'boolean' &&
+                !hit['permissions']['canEdit']) ||
+                (typeof this.regions_['is_modified'] === 'boolean' &&
+                this.regions_['is_modified']))
+                    return null;
+        return hit;
+    };
+
+    /**
+     * a sort of desctructor
```
Comment (jburel) [inline src/viewers/viewer/interaction/Translate.js]: typo

File: src/viewers/viewer/tiles/ImageTile.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        this.imageByContext_ = {};
+    };
+
+    /**
+     * A convenience method to draw the tile into the context to then be called
+     * within the scope of the post tile load hook for example...
+     *
+     * @function
+     * @private
+     * @param {Object} tile the tile as an img object
```
Comment (jburel) [inline src/viewers/viewer/tiles/ImageTile.js]: image

File: src/viewers/viewer/tiles/ImageTile.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+     * @param {Array.<number>} tileSize the tile size as an array [width, height]
+     * @param {boolean=} createContextOnlyForResize renders on canvas only if resize is needed
```
Comment (jburel) [inline src/viewers/viewer/tiles/ImageTile.js]: ``=``

File: src/viewers/viewer/tiles/ImageTile.js [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+     * @param {string=} key the uid (for createContextOnlyForResize is true)
```
Comment (jburel) [inline src/viewers/viewer/tiles/ImageTile.js]: same

File: src/viewers/viewer/tiles/ImageTile.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        if (createContextOnlyForResize && key)
+            this.imageByContext_[key] = context.canvas;
+
+        return context.canvas;
+    }
+
+    /**
+     * Get the HTML image element for this tile (may be a Canvas, Image, or Video).
+     * @function
+     * @param {Object=} opt_context Object.
```
Comment (jburel) [inline src/viewers/viewer/tiles/ImageTile.js]: same

File: src/viewers/viewer/interaction/Select.js [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+            feature['selected'] = true;
+            this.getFeatures().push(feature);
+        } else {
+            feature['selected'] = false;
+            this.getFeatures().remove(feature);
+        }
+    }
+
+    /**
+     * a sort of desctructor
```
Comment (jburel) [inline src/viewers/viewer/interaction/Select.js]: typo

File: src/viewers/viewer/interaction/BoxSelect.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    /**
+     * Unregister the start/end listeners
+     */
+    unregisterListeners() {
+        if (this.boxStartListener_) unlistenByKey(this.boxStartListener_);
+        if (this.boxEndListener_) unlistenByKey(this.boxEndListener_);
+    }
+
+    /**
+     * a sort of desctructor
```
Comment (jburel) [inline src/viewers/viewer/interaction/BoxSelect.js]: typo

File: plugin/setup.py (hunk 1 above)
Comment (will-moore) [reply plugin/setup.py]: This and other TODOs moved to https://trello.com/c/vyZSm0fj/15-openlayers-5-build-cleanup

**Comment by jburel** [comment]
Merging so we can move forward with the rest of the work

**Pull request merged**